include/hal/library/cryptlib/cryptlib_ec.h

```diff
@@ -30,6 +30,24 @@ extern void *libspdm_ec_new_by_nid(size_t nid);
 extern void libspdm_ec_free(void *ec_context);
 #endif /* (LIBSPDM_ECDHE_SUPPORT) || (LIBSPDM_ECDSA_SUPPORT) */
 
+/**
+ * Sets the private key component into the established EC context.
+ *
+ * For P-256, the private_key_size is 32 byte.
+ * For P-384, the private_key_size is 48 byte.
+ * For P-521, the private_key_size is 66 byte.
+ *
+ * @param[in, out]  ec_context       Pointer to EC context being set.
+ * @param[in]       private_key      Pointer to the private key buffer.
+ * @param[in]       private_key_size The size of private key buffer in bytes.
+ *
+ * @retval  true   EC private key component was set successfully.
+ * @retval  false  Invalid EC private key component.
+ *
+ **/
+extern bool libspdm_ec_set_priv_key(void *ec_context, const uint8_t *private_key,
```

Comment:
This needs to be behind (at least) `LIBSPDM_ECDSA_SUPPORT` and `LIBSPDM_FIPS_MODE`.

Reply:
In C it's legal to have multiple function prototype declarations within a translation unit, so let's add `#if LIBSPDM_FIPS_MODE` around this function here and restore the declaration in `os_stub/spdm_crypt_ext_lib/cryptlib_ext.h`

Reply:
Thanks. That's a good idea. I have updated the code.